manifests/interface.pp

```diff
@@ -222,208 +229,212 @@
 #     number (portno) of qeth devices.
 define network::interface (
 
-  $enable                = true,
-  $ensure                = 'present',
-  $template              = "network/interface/${::osfamily}.erb",
-  $options               = undef,
-  $interface             = $name,
-  $restart_all_nic       = true,
+  $enable                   = true,
+  $ensure                   = 'present',
+  $template                 = "network/interface/${::osfamily}.erb",
+  $options                  = undef,
+  $interface                = $name,
+  $restart_all_nic          = true,
+  $interface_reload_command = $::operatingsystem ? {
+                                'CumulusLinux' => 'ifreload -a',
+                                default        => "ifdown ${name}; ifup ${name}",
+                              }
```

Comment:
you need a comma here.
Also I don't like selectors in class params, you can add a reload_command parameter (we are already managing the interface, so no need to write interface_ in the param) which by efault is undef , then set and use a variable like:
```
$real_reload_command = $reload_command ? {
  undef => $::operatingsystem ? {
    'CumulusLinux' => 'ifreload -a',
    default        => "ifdown ${name}; ifup ${name}",
  },
  default => $reload_command,
}
```

Then, of course, use $real_reload_command instead of $interface_reload_command in exec { "network_restart_${name}":

Reply:
Sorry about the comma, I should've caught that.
You're right moving the selector out of the parameter section is cleaner. I'll make the changes and test it.